index-configuration-tool/main.py

```diff
@@ -0,0 +1,135 @@
+import sys
+import requests
+import logstash_conf_parser as parser
+from jsondiff import diff
+
+# Constants
+SUPPORTED_ENDPOINTS = ["opensearch", "elasticsearch"]
+HOSTS_KEY = "hosts"
+USER_KEY = "user"
+PWD_KEY = "password"
+SETTINGS_KEY = "settings"
+MAPPINGS_KEY = "mappings"
+INDEX_KEY = "index"
+INTERNAL_SETTINGS_KEYS = ["creation_date", "uuid", "provided_name", "version", "store"]
+ALL_INDICES_ENDPOINT = "*"
+
+
+# Utility method to make a comma-separated string from a set
+def string_from_set(s: set) -> str:
+    if s:
+        return ", ".join(s)
+    else:
+        return "[]"
+
+
+# Utility method to get the plugin config from the base data
+def get_plugin_config(data: dict, plugin_type: str) -> dict:
+    # Tuple contents are (type, config)
+    return data[plugin_type][1]
+
+
+# TODO Only supports basic auth for now
+def get_auth(input_data: dict) -> tuple:
+    if USER_KEY in input_data and PWD_KEY in input_data:
+        return input_data[USER_KEY], input_data[PWD_KEY]
+
+
+def get_endpoint_info(plugin_config: dict) -> tuple:
+    endpoint = "https://" if ("ssl" in plugin_config and plugin_config["ssl"]) else "http://"
+    endpoint += plugin_config[HOSTS_KEY][0] if type(plugin_config[HOSTS_KEY]) is list else plugin_config[HOSTS_KEY]
```

Comment:
When would this be a list?  Should you just take the first element or should you throw an error / make the user choose somehow.